File: src/etch/bindings/csharp/msg/test/TestIdName.cs
///$Id$
///
///Created by Champakesan, Badri Narayanan on Jun 8, 2007.
///
///Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.


using System;
using NUnit.Framework;

namespace Etch.Msg.Test
{

    [TestFixture]
    public class TestIdName
    {
        [Test]
        public void Id()
        {
            Assert.AreEqual(1, new IdName(1, "foo").Id);
            Assert.AreEqual(1, new IdName(1, "bar").Id);
            Assert.AreEqual(2, new IdName(2, "foo").Id);
            Assert.AreEqual(2, new IdName(2, "bar").Id);
        }

        [Test]
        public void Name()
        {
            Assert.AreEqual("foo", new IdName(1, "foo").Name);
            Assert.AreEqual("bar", new IdName(1, "bar").Name);
            Assert.AreEqual("foo", new IdName(2, "foo").Name);
            Assert.AreEqual("bar", new IdName(2, "bar").Name);
        }

        [Test]
        public void XToString() 
        {
            Assert.AreEqual( "foo(1)", new IdName( 1, "foo" ).ToString() );
            Assert.AreEqual( "bar(1)", new IdName( 1, "bar" ).ToString() );
            Assert.AreEqual( "foo(2)", new IdName( 2, "foo" ).ToString() );
            Assert.AreEqual( "bar(2)", new IdName( 2, "bar" ).ToString() );
        }

        [Test]
        public void EqualsObject()
        {
            Assert.IsTrue( new IdName( 1, "foo" ).Equals( new IdName( 1, "foo" ) ) );
            Assert.IsTrue( new IdName( 1, "bar" ).Equals( new IdName( 1, "bar" ) ) );
            Assert.IsTrue( new IdName( 2, "foo" ).Equals( new IdName( 2, "foo" ) ) );
            Assert.IsTrue( new IdName( 2, "bar" ).Equals( new IdName( 2, "bar" ) ) );

            Assert.IsFalse( new IdName( 1, "foo" ).Equals( new IdName( 2, "foo" ) ) ); // id changed
            Assert.IsFalse( new IdName( 1, "foo" ).Equals( new IdName( 1, "bar" ) ) ); // name changed
            Assert.IsFalse( new IdName( 1, "foo" ).Equals( new IdName( 2, "bar" ) ) ); // id and name changed

            Assert.IsFalse( new IdName( 1, "foo" ).Equals( null ) );
            Assert.IsFalse( new IdName( 1, "foo" ).Equals( 1 ) );
            Assert.IsFalse( new IdName( 1, "foo" ).Equals( true ) );
        }

        [Test]
        public void HashCodeObject() 
        {
            // you cannot really predict when HashCode will not be the same. the
            // rule is that it must be the same when the objects would compare equal.
            Assert.AreEqual( new IdName( 1, "foo" ).GetHashCode(), new IdName( 1, "foo" ).GetHashCode() );
            Assert.AreEqual( new IdName( 2, "foo" ).GetHashCode(), new IdName( 2, "foo" ).GetHashCode() );
            Assert.AreEqual( new IdName( 1, "bar" ).GetHashCode(), new IdName( 1, "bar" ).GetHashCode() );
            Assert.AreEqual( new IdName( 2, "bar" ).GetHashCode(), new IdName( 2, "bar" ).GetHashCode() );
        }

        [Test]
        public void Hash()
        {
            Assert.AreEqual(5381, IdName.Hash(""));
            Assert.AreEqual(352988316, IdName.Hash("a"));
            Assert.AreEqual(1511848646, IdName.Hash("ab"));
            Assert.AreEqual(669497117, IdName.Hash("abc"));
            Assert.AreEqual(-1994190713, IdName.Hash("abcd"));
            Assert.AreEqual(-802680418, IdName.Hash("abcde"));
            Assert.AreEqual(1266308680, IdName.Hash("abcdef"));
            Assert.AreEqual(-379372513, IdName.Hash("abcdefg"));
            Assert.AreEqual(-1416967159, IdName.Hash("abcdefgh"));
            Assert.AreEqual(53556896, IdName.Hash("abcdefghi"));
            Assert.AreEqual(-4427318, IdName.Hash("abcdefghij"));
        }
    }
}

File: src/etch/bindings/java/msg/test/TestIdName.java
/* $Id$
 *
 * Created by sccomer on Feb 4, 2007.
 *
 * Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.
 */

package etch.bindings.java.msg.test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

import etch.bindings.java.msg.IdName;

/** */
public class TestIdName
{
	/** */
	@Test
	public void getId()
	{
		assertEquals( 1, new IdName( 1, "foo" ).getId() );
		assertEquals( 1, new IdName( 1, "bar" ).getId() );
		assertEquals( 2, new IdName( 2, "foo" ).getId() );
		assertEquals( 2, new IdName( 2, "bar" ).getId() );
	}

	/** */
	@Test
	public void getName()
	{
		assertEquals( "foo", new IdName( 1, "foo" ).getName() );
		assertEquals( "bar", new IdName( 1, "bar" ).getName() );
		assertEquals( "foo", new IdName( 2, "foo" ).getName() );
		assertEquals( "bar", new IdName( 2, "bar" ).getName() );
	}

	/** */
	@Test
	public void xtoString()
	{
		assertEquals( "foo", new IdName( 1, "foo" ).toString() );
		assertEquals( "bar", new IdName( 1, "bar" ).toString() );
		assertEquals( "foo", new IdName( 2, "foo" ).toString() );
		assertEquals( "bar", new IdName( 2, "bar" ).toString() );
	}

	/** */
	@Test
	public void equalsObject()
	{
		assertTrue( new IdName( 1, "foo" ).equals( new IdName( 1, "foo" ) ) );
		assertTrue( new IdName( 1, "foo" ).equals( new IdName( 1, "bar" ) ) );
		assertTrue( new IdName( 2, "foo" ).equals( new IdName( 2, "foo" ) ) );
		assertTrue( new IdName( 2, "foo" ).equals( new IdName( 2, "bar" ) ) );
		
		assertFalse( new IdName( 1, "foo" ).equals( new IdName( 2, "foo" ) ) );
		assertFalse( new IdName( 1, "foo" ).equals( new IdName( 2, "bar" ) ) );
		assertFalse( new IdName( 2, "foo" ).equals( new IdName( 1, "foo" ) ) );
		assertFalse( new IdName( 2, "foo" ).equals( new IdName( 1, "bar" ) ) );
		
		assertFalse( new IdName( 1, "foo" ).equals( null ) );
		assertFalse( new IdName( 1, "foo" ).equals( 1 ) );
		assertFalse( new IdName( 1, "foo" ).equals( true ) );
	}

	/** */
	@Test
	public void hashCodeObject()
	{
		// you cannot really predict when hashCode will not be the same. the
		// rule is that it must be the same when the objects would compare equal.
		assertEquals( new IdName( 1, "foo" ).hashCode(), new IdName( 1, "foo" ).hashCode() );
		assertEquals( new IdName( 1, "foo" ).hashCode(), new IdName( 1, "bar" ).hashCode() );
		assertEquals( new IdName( 2, "foo" ).hashCode(), new IdName( 2, "foo" ).hashCode() );
		assertEquals( new IdName( 2, "foo" ).hashCode(), new IdName( 2, "bar" ).hashCode() );
	}

	/** */
	@Test
	public void hash()
	{
		assertEquals( 5381, IdName.hash( "" ) );
		assertEquals( 352988316, IdName.hash( "a" ) );
		assertEquals( 1511848646, IdName.hash( "ab" ) );
		assertEquals( 669497117, IdName.hash( "abc" ) );
		assertEquals( -1994190713, IdName.hash( "abcd" ) );
		assertEquals( -802680418, IdName.hash( "abcde" ) );
		assertEquals( 1266308680, IdName.hash( "abcdef" ) );
		assertEquals( -379372513, IdName.hash( "abcdefg" ) );
		assertEquals( -1416967159, IdName.hash( "abcdefgh" ) );
		assertEquals( 53556896, IdName.hash( "abcdefghi" ) );
		assertEquals( -4427318, IdName.hash( "abcdefghij" ) );
	}
}
